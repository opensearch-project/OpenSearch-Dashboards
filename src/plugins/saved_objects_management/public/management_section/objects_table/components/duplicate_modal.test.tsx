/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  httpServiceMock,
  notificationServiceMock,
  workspacesServiceMock,
} from '../../../../../../core/public/mocks';
import { ShowDuplicateModalProps, SavedObjectsDuplicateModal } from './duplicate_modal';
import { SavedObjectWithMetadata } from '../../../types';
import { shallowWithI18nProvider } from 'test_utils/enzyme_helpers';
import React from 'react';
import { WorkspaceAttribute } from 'src/core/types';
import { render } from '@testing-library/react';
import { WorkspaceOption } from './utils';

interface Props extends ShowDuplicateModalProps {
  onClose: () => void;
}

describe('DuplicateModal', () => {
  let duplicateProps: Props;
  let http: ReturnType<typeof httpServiceMock.createStartContract>;
  let notifications: ReturnType<typeof notificationServiceMock.createStartContract>;
  let workspaces: ReturnType<typeof workspacesServiceMock.createStartContract>;
  const selectedSavedObjects: SavedObjectWithMetadata[] = [
    {
      id: '1',
      type: 'dashboard',
      workspaces: ['workspace1'],
      attributes: {},
      references: [],
      meta: {
        title: 'Dashboard_1',
        icon: 'dashboardApp',
      },
    },
    {
      id: '2',
      type: 'visualization',
      workspaces: ['workspace2'],
      attributes: {},
      references: [],
      meta: {
        title: 'Visualization',
        icon: 'visualizationApp',
      },
    },
    {
      id: '3',
      type: 'dashboard',
      workspaces: ['workspace2'],
      attributes: {},
      references: [],
      meta: {
        title: 'Dashboard_2',
      },
    },
  ];
  const workspaceList: WorkspaceAttribute[] = [
    {
      id: 'workspace1',
      name: 'foo',
    },
    {
      id: 'workspace2',
      name: 'bar',
    },
  ];
  beforeEach(() => {
    http = httpServiceMock.createStartContract();
    notifications = notificationServiceMock.createStartContract();
    workspaces = workspacesServiceMock.createStartContract();

    duplicateProps = {
      onDuplicate: jest.fn(),
      onClose: jest.fn(),
      http,
      workspaces,
      notifications,
      selectedSavedObjects,
    };
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render normally', async () => {
    render(<SavedObjectsDuplicateModal {...duplicateProps} />);
    expect(document.children).toMatchSnapshot();
  });

  it('should Unmount normally', async () => {
    const component = shallowWithI18nProvider(<SavedObjectsDuplicateModal {...duplicateProps} />);
    expect(component.unmount()).toMatchSnapshot();
  });

  it('should show all target workspace options when not in any workspace', async () => {
    workspaces.workspaceList$.next(workspaceList);
    workspaces.currentWorkspaceId$.next('');
    workspaces.currentWorkspace$.next(null);
    duplicateProps = { ...duplicateProps, workspaces };
    const component = shallowWithI18nProvider(<SavedObjectsDuplicateModal {...duplicateProps} />);
    await new Promise((resolve) => process.nextTick(resolve));
    component.update();
    const options = component.find('EuiComboBox').prop('options') as WorkspaceOption[];
    expect(options.length).toEqual(2);
    expect(options[0].label).toEqual('foo');
    expect(options[1].label).toEqual('bar');
  });

  it('should display the suffix (current) in target workspace options when it is the current workspace', async () => {
    workspaces.workspaceList$.next(workspaceList);
    workspaces.currentWorkspaceId$.next('workspace1');
    workspaces.currentWorkspace$.next({
      id: 'workspace1',
      name: 'foo',
    });
    duplicateProps = { ...duplicateProps, workspaces };
    const component = shallowWithI18nProvider(<SavedObjectsDuplicateModal {...duplicateProps} />);
    await new Promise((resolve) => process.nextTick(resolve));
    component.update();
    const options = component.find('EuiComboBox').prop('options') as WorkspaceOption[];
    expect(options.length).toEqual(2);
    expect(options[0].label).toEqual('foo (current)');
    expect(options[1].label).toEqual('bar');
  });

  it('should uncheck duplicate related objects', async () => {
    const component = shallowWithI18nProvider(<SavedObjectsDuplicateModal {...duplicateProps} />);

    const euiCheckbox = component.find('EuiCheckbox').at(0);
    expect(euiCheckbox.prop('checked')).toEqual(true);
    expect(euiCheckbox.prop('id')).toEqual('includeReferencesDeep');
    expect(component.state('isIncludeReferencesDeepChecked')).toEqual(true);

    euiCheckbox.simulate('change', { target: { checked: false } });
    expect(component.state('isIncludeReferencesDeepChecked')).toEqual(false);
  });

  it('should call onClose function when cancle button is clicked', () => {
    const component = shallowWithI18nProvider(<SavedObjectsDuplicateModal {...duplicateProps} />);
    component.find('[data-test-subj="duplicateCancelButton"]').simulate('click');
    expect(duplicateProps.onClose).toHaveBeenCalled();
  });

  it('should call onDuplicate function when confirm button is clicked', () => {
    workspaces.workspaceList$.next(workspaceList);
    workspaces.currentWorkspaceId$.next('');
    workspaces.currentWorkspace$.next(null);
    duplicateProps = { ...duplicateProps, workspaces };
    const component = shallowWithI18nProvider(<SavedObjectsDuplicateModal {...duplicateProps} />);
    const comboBox = component.find('EuiComboBox');
    comboBox.simulate('change', [{ label: 'bar', key: 'workspace2', value: workspaceList[1] }]);
    const confirmButton = component.find('[data-test-subj="duplicateConfirmButton"]');
    expect(confirmButton.prop('isLoading')).toBe(false);
    expect(confirmButton.prop('disabled')).toBe(false);
    confirmButton.simulate('click');
    expect(duplicateProps.onDuplicate).toHaveBeenCalled();
  });

  it('should not change isLoading when isMounted is false ', async () => {
    const component = shallowWithI18nProvider(<SavedObjectsDuplicateModal {...duplicateProps} />);
    const comboBox = component.find('EuiComboBox');
    comboBox.simulate('change', [{ label: 'bar', key: 'workspace2', value: workspaceList[1] }]);
    const confirmButton = component.find('[data-test-subj="duplicateConfirmButton"]');
    (component.instance() as any).isMounted = false;
    confirmButton.simulate('click');
    expect(duplicateProps.onDuplicate).toHaveBeenCalled();
    expect(component.state('isLoading')).toBe(true);
  });
});
