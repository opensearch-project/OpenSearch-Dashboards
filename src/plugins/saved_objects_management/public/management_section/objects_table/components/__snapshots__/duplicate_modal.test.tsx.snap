// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DuplicateModal should Unmount normally 1`] = `""`;

exports[`DuplicateModal should render normally 1`] = `
HTMLCollection [
  <html>
    <head>
      <style
        id="ace_editor.css"
      >
        .ace_editor {position: relative;overflow: hidden;font: 12px/normal 'Monaco', 'Menlo', 'Ubuntu Mono', 'Consolas', 'source-code-pro', monospace;direction: ltr;text-align: left;-webkit-tap-highlight-color: rgba(0, 0, 0, 0);}.ace_scroller {position: absolute;overflow: hidden;top: 0;bottom: 0;background-color: inherit;-ms-user-select: none;-moz-user-select: none;-webkit-user-select: none;user-select: none;cursor: text;}.ace_content {position: absolute;-moz-box-sizing: border-box;-webkit-box-sizing: border-box;box-sizing: border-box;min-width: 100%;}.ace_dragging .ace_scroller:before{position: absolute;top: 0;left: 0;right: 0;bottom: 0;content: '';background: rgba(250, 250, 250, 0.01);z-index: 1000;}.ace_dragging.ace_dark .ace_scroller:before{background: rgba(0, 0, 0, 0.01);}.ace_selecting, .ace_selecting * {cursor: text !important;}.ace_gutter {position: absolute;overflow : hidden;width: auto;top: 0;bottom: 0;left: 0;cursor: default;z-index: 4;-ms-user-select: none;-moz-user-select: none;-webkit-user-select: none;user-select: none;}.ace_gutter-active-line {position: absolute;left: 0;right: 0;}.ace_scroller.ace_scroll-left {box-shadow: 17px 0 16px -16px rgba(0, 0, 0, 0.4) inset;}.ace_gutter-cell {padding-left: 19px;padding-right: 6px;background-repeat: no-repeat;}.ace_gutter-cell.ace_error {background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAMAAAAoLQ9TAAABOFBMVEX/////////QRswFAb/Ui4wFAYwFAYwFAaWGAfDRymzOSH/PxswFAb/SiUwFAYwFAbUPRvjQiDllog5HhHdRybsTi3/Tyv9Tir+Syj/UC3////XurebMBIwFAb/RSHbPx/gUzfdwL3kzMivKBAwFAbbvbnhPx66NhowFAYwFAaZJg8wFAaxKBDZurf/RB6mMxb/SCMwFAYwFAbxQB3+RB4wFAb/Qhy4Oh+4QifbNRcwFAYwFAYwFAb/QRzdNhgwFAYwFAbav7v/Uy7oaE68MBK5LxLewr/r2NXewLswFAaxJw4wFAbkPRy2PyYwFAaxKhLm1tMwFAazPiQwFAaUGAb/QBrfOx3bvrv/VC/maE4wFAbRPBq6MRO8Qynew8Dp2tjfwb0wFAbx6eju5+by6uns4uH9/f36+vr/GkHjAAAAYnRSTlMAGt+64rnWu/bo8eAA4InH3+DwoN7j4eLi4xP99Nfg4+b+/u9B/eDs1MD1mO7+4PHg2MXa347g7vDizMLN4eG+Pv7i5evs/v79yu7S3/DV7/498Yv24eH+4ufQ3Ozu/v7+y13sRqwAAADLSURBVHjaZc/XDsFgGIBhtDrshlitmk2IrbHFqL2pvXf/+78DPokj7+Fz9qpU/9UXJIlhmPaTaQ6QPaz0mm+5gwkgovcV6GZzd5JtCQwgsxoHOvJO15kleRLAnMgHFIESUEPmawB9ngmelTtipwwfASilxOLyiV5UVUyVAfbG0cCPHig+GBkzAENHS0AstVF6bacZIOzgLmxsHbt2OecNgJC83JERmePUYq8ARGkJx6XtFsdddBQgZE2nPR6CICZhawjA4Fb/chv+399kfR+MMMDGOQAAAABJRU5ErkJggg==");background-repeat: no-repeat;background-position: 2px center;}.ace_gutter-cell.ace_warning {background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAMAAAAoLQ9TAAAAmVBMVEX///8AAAD///8AAAAAAABPSzb/5sAAAAB/blH/73z/ulkAAAAAAAD85pkAAAAAAAACAgP/vGz/rkDerGbGrV7/pkQICAf////e0IsAAAD/oED/qTvhrnUAAAD/yHD/njcAAADuv2r/nz//oTj/p064oGf/zHAAAAA9Nir/tFIAAAD/tlTiuWf/tkIAAACynXEAAAAAAAAtIRW7zBpBAAAAM3RSTlMAABR1m7RXO8Ln31Z36zT+neXe5OzooRDfn+TZ4p3h2hTf4t3k3ucyrN1K5+Xaks52Sfs9CXgrAAAAjklEQVR42o3PbQ+CIBQFYEwboPhSYgoYunIqqLn6/z8uYdH8Vmdnu9vz4WwXgN/xTPRD2+sgOcZjsge/whXZgUaYYvT8QnuJaUrjrHUQreGczuEafQCO/SJTufTbroWsPgsllVhq3wJEk2jUSzX3CUEDJC84707djRc5MTAQxoLgupWRwW6UB5fS++NV8AbOZgnsC7BpEAAAAABJRU5ErkJggg==");background-position: 2px center;}.ace_gutter-cell.ace_info {background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAAAAAA6mKC9AAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAAJ0Uk5TAAB2k804AAAAPklEQVQY02NgIB68QuO3tiLznjAwpKTgNyDbMegwisCHZUETUZV0ZqOquBpXj2rtnpSJT1AEnnRmL2OgGgAAIKkRQap2htgAAAAASUVORK5CYII=");background-position: 2px center;}.ace_dark .ace_gutter-cell.ace_info {background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQBAMAAADt3eJSAAAAJFBMVEUAAAChoaGAgIAqKiq+vr6tra1ZWVmUlJSbm5s8PDxubm56enrdgzg3AAAAAXRSTlMAQObYZgAAAClJREFUeNpjYMAPdsMYHegyJZFQBlsUlMFVCWUYKkAZMxZAGdxlDMQBAG+TBP4B6RyJAAAAAElFTkSuQmCC");}.ace_scrollbar {position: absolute;right: 0;bottom: 0;z-index: 6;}.ace_scrollbar-inner {position: absolute;cursor: text;left: 0;top: 0;}.ace_scrollbar-v{overflow-x: hidden;overflow-y: scroll;top: 0;}.ace_scrollbar-h {overflow-x: scroll;overflow-y: hidden;left: 0;}.ace_print-margin {position: absolute;height: 100%;}.ace_text-input {position: absolute;z-index: 0;width: 0.5em;height: 1em;opacity: 0;background: transparent;-moz-appearance: none;appearance: none;border: none;resize: none;outline: none;overflow: hidden;font: inherit;padding: 0 1px;margin: 0 -1px;text-indent: -1em;-ms-user-select: text;-moz-user-select: text;-webkit-user-select: text;user-select: text;white-space: pre!important;}.ace_text-input.ace_composition {background: inherit;color: inherit;z-index: 1000;opacity: 1;text-indent: 0;}.ace_layer {z-index: 1;position: absolute;overflow: hidden;word-wrap: normal;white-space: pre;height: 100%;width: 100%;-moz-box-sizing: border-box;-webkit-box-sizing: border-box;box-sizing: border-box;pointer-events: none;}.ace_gutter-layer {position: relative;width: auto;text-align: right;pointer-events: auto;}.ace_text-layer {font: inherit !important;}.ace_cjk {display: inline-block;text-align: center;}.ace_cursor-layer {z-index: 4;}.ace_cursor {z-index: 4;position: absolute;-moz-box-sizing: border-box;-webkit-box-sizing: border-box;box-sizing: border-box;border-left: 2px solid;transform: translatez(0);}.ace_multiselect .ace_cursor {border-left-width: 1px;}.ace_slim-cursors .ace_cursor {border-left-width: 1px;}.ace_overwrite-cursors .ace_cursor {border-left-width: 0;border-bottom: 1px solid;}.ace_hidden-cursors .ace_cursor {opacity: 0.2;}.ace_smooth-blinking .ace_cursor {-webkit-transition: opacity 0.18s;transition: opacity 0.18s;}.ace_marker-layer .ace_step, .ace_marker-layer .ace_stack {position: absolute;z-index: 3;}.ace_marker-layer .ace_selection {position: absolute;z-index: 5;}.ace_marker-layer .ace_bracket {position: absolute;z-index: 6;}.ace_marker-layer .ace_active-line {position: absolute;z-index: 2;}.ace_marker-layer .ace_selected-word {position: absolute;z-index: 4;-moz-box-sizing: border-box;-webkit-box-sizing: border-box;box-sizing: border-box;}.ace_line .ace_fold {-moz-box-sizing: border-box;-webkit-box-sizing: border-box;box-sizing: border-box;display: inline-block;height: 11px;margin-top: -2px;vertical-align: middle;background-image:url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABEAAAAJCAYAAADU6McMAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAJpJREFUeNpi/P//PwOlgAXGYGRklAVSokD8GmjwY1wasKljQpYACtpCFeADcHVQfQyMQAwzwAZI3wJKvCLkfKBaMSClBlR7BOQikCFGQEErIH0VqkabiGCAqwUadAzZJRxQr/0gwiXIal8zQQPnNVTgJ1TdawL0T5gBIP1MUJNhBv2HKoQHHjqNrA4WO4zY0glyNKLT2KIfIMAAQsdgGiXvgnYAAAAASUVORK5CYII="),url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAA3CAYAAADNNiA5AAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAACJJREFUeNpi+P//fxgTAwPDBxDxD078RSX+YeEyDFMCIMAAI3INmXiwf2YAAAAASUVORK5CYII=");background-repeat: no-repeat, repeat-x;background-position: center center, top left;color: transparent;border: 1px solid black;border-radius: 2px;cursor: pointer;pointer-events: auto;}.ace_dark .ace_fold {}.ace_fold:hover{background-image:url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABEAAAAJCAYAAADU6McMAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAJpJREFUeNpi/P//PwOlgAXGYGRklAVSokD8GmjwY1wasKljQpYACtpCFeADcHVQfQyMQAwzwAZI3wJKvCLkfKBaMSClBlR7BOQikCFGQEErIH0VqkabiGCAqwUadAzZJRxQr/0gwiXIal8zQQPnNVTgJ1TdawL0T5gBIP1MUJNhBv2HKoQHHjqNrA4WO4zY0glyNKLT2KIfIMAAQsdgGiXvgnYAAAAASUVORK5CYII="),url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAA3CAYAAADNNiA5AAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAACBJREFUeNpi+P//fz4TAwPDZxDxD5X4i5fLMEwJgAADAEPVDbjNw87ZAAAAAElFTkSuQmCC");}.ace_tooltip {background-color: #FFF;background-image: -webkit-linear-gradient(top, transparent, rgba(0, 0, 0, 0.1));background-image: linear-gradient(to bottom, transparent, rgba(0, 0, 0, 0.1));border: 1px solid gray;border-radius: 1px;box-shadow: 0 1px 2px rgba(0, 0, 0, 0.3);color: black;max-width: 100%;padding: 3px 4px;position: fixed;z-index: 999999;-moz-box-sizing: border-box;-webkit-box-sizing: border-box;box-sizing: border-box;cursor: default;white-space: pre;word-wrap: break-word;line-height: normal;font-style: normal;font-weight: normal;letter-spacing: normal;pointer-events: none;}.ace_folding-enabled &gt; .ace_gutter-cell {padding-right: 13px;}.ace_fold-widget {-moz-box-sizing: border-box;-webkit-box-sizing: border-box;box-sizing: border-box;margin: 0 -12px 0 1px;display: none;width: 11px;vertical-align: top;background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAANElEQVR42mWKsQ0AMAzC8ixLlrzQjzmBiEjp0A6WwBCSPgKAXoLkqSot7nN3yMwR7pZ32NzpKkVoDBUxKAAAAABJRU5ErkJggg==");background-repeat: no-repeat;background-position: center;border-radius: 3px;border: 1px solid transparent;cursor: pointer;}.ace_folding-enabled .ace_fold-widget {display: inline-block;   }.ace_fold-widget.ace_end {background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAANElEQVR42m3HwQkAMAhD0YzsRchFKI7sAikeWkrxwScEB0nh5e7KTPWimZki4tYfVbX+MNl4pyZXejUO1QAAAABJRU5ErkJggg==");}.ace_fold-widget.ace_closed {background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAMAAAAGCAYAAAAG5SQMAAAAOUlEQVR42jXKwQkAMAgDwKwqKD4EwQ26sSOkVWjgIIHAzPiCgaqiqnJHZnKICBERHN194O5b9vbLuAVRL+l0YWnZAAAAAElFTkSuQmCCXA==");}.ace_fold-widget:hover {border: 1px solid rgba(0, 0, 0, 0.3);background-color: rgba(255, 255, 255, 0.2);box-shadow: 0 1px 1px rgba(255, 255, 255, 0.7);}.ace_fold-widget:active {border: 1px solid rgba(0, 0, 0, 0.4);background-color: rgba(0, 0, 0, 0.05);box-shadow: 0 1px 1px rgba(255, 255, 255, 0.8);}.ace_dark .ace_fold-widget {background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHklEQVQIW2P4//8/AzoGEQ7oGCaLLAhWiSwB146BAQCSTPYocqT0AAAAAElFTkSuQmCC");}.ace_dark .ace_fold-widget.ace_end {background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAH0lEQVQIW2P4//8/AxQ7wNjIAjDMgC4AxjCVKBirIAAF0kz2rlhxpAAAAABJRU5ErkJggg==");}.ace_dark .ace_fold-widget.ace_closed {background-image: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAMAAAAFCAYAAACAcVaiAAAAHElEQVQIW2P4//+/AxAzgDADlOOAznHAKgPWAwARji8UIDTfQQAAAABJRU5ErkJggg==");}.ace_dark .ace_fold-widget:hover {box-shadow: 0 1px 1px rgba(255, 255, 255, 0.2);background-color: rgba(255, 255, 255, 0.1);}.ace_dark .ace_fold-widget:active {box-shadow: 0 1px 1px rgba(255, 255, 255, 0.2);}.ace_fold-widget.ace_invalid {background-color: #FFB4B4;border-color: #DE5555;}.ace_fade-fold-widgets .ace_fold-widget {-webkit-transition: opacity 0.4s ease 0.05s;transition: opacity 0.4s ease 0.05s;opacity: 0;}.ace_fade-fold-widgets:hover .ace_fold-widget {-webkit-transition: opacity 0.05s ease 0.05s;transition: opacity 0.05s ease 0.05s;opacity:1;}.ace_underline {text-decoration: underline;}.ace_bold {font-weight: bold;}.ace_nobold .ace_bold {font-weight: normal;}.ace_italic {font-style: italic;}.ace_error-marker {background-color: rgba(255, 0, 0,0.2);position: absolute;z-index: 9;}.ace_highlight-marker {background-color: rgba(255, 255, 0,0.2);position: absolute;z-index: 8;}.ace_br1 {border-top-left-radius    : 3px;}.ace_br2 {border-top-right-radius   : 3px;}.ace_br3 {border-top-left-radius    : 3px; border-top-right-radius:    3px;}.ace_br4 {border-bottom-right-radius: 3px;}.ace_br5 {border-top-left-radius    : 3px; border-bottom-right-radius: 3px;}.ace_br6 {border-top-right-radius   : 3px; border-bottom-right-radius: 3px;}.ace_br7 {border-top-left-radius    : 3px; border-top-right-radius:    3px; border-bottom-right-radius: 3px;}.ace_br8 {border-bottom-left-radius : 3px;}.ace_br9 {border-top-left-radius    : 3px; border-bottom-left-radius:  3px;}.ace_br10{border-top-right-radius   : 3px; border-bottom-left-radius:  3px;}.ace_br11{border-top-left-radius    : 3px; border-top-right-radius:    3px; border-bottom-left-radius:  3px;}.ace_br12{border-bottom-right-radius: 3px; border-bottom-left-radius:  3px;}.ace_br13{border-top-left-radius    : 3px; border-bottom-right-radius: 3px; border-bottom-left-radius:  3px;}.ace_br14{border-top-right-radius   : 3px; border-bottom-right-radius: 3px; border-bottom-left-radius:  3px;}.ace_br15{border-top-left-radius    : 3px; border-top-right-radius:    3px; border-bottom-right-radius: 3px; border-bottom-left-radius: 3px;}.ace_text-input-ios {position: absolute !important;top: -100000px !important;left: -100000px !important;}
/*# sourceURL=ace/css/ace_editor.css */
      </style>
      <style
        id="ace-tm"
      >
        .ace-tm .ace_gutter {background: #f0f0f0;color: #333;}.ace-tm .ace_print-margin {width: 1px;background: #e8e8e8;}.ace-tm .ace_fold {background-color: #6B72E6;}.ace-tm {background-color: #FFFFFF;color: black;}.ace-tm .ace_cursor {color: black;}.ace-tm .ace_invisible {color: rgb(191, 191, 191);}.ace-tm .ace_storage,.ace-tm .ace_keyword {color: blue;}.ace-tm .ace_constant {color: rgb(197, 6, 11);}.ace-tm .ace_constant.ace_buildin {color: rgb(88, 72, 246);}.ace-tm .ace_constant.ace_language {color: rgb(88, 92, 246);}.ace-tm .ace_constant.ace_library {color: rgb(6, 150, 14);}.ace-tm .ace_invalid {background-color: rgba(255, 0, 0, 0.1);color: red;}.ace-tm .ace_support.ace_function {color: rgb(60, 76, 114);}.ace-tm .ace_support.ace_constant {color: rgb(6, 150, 14);}.ace-tm .ace_support.ace_type,.ace-tm .ace_support.ace_class {color: rgb(109, 121, 222);}.ace-tm .ace_keyword.ace_operator {color: rgb(104, 118, 135);}.ace-tm .ace_string {color: rgb(3, 106, 7);}.ace-tm .ace_comment {color: rgb(76, 136, 107);}.ace-tm .ace_comment.ace_doc {color: rgb(0, 102, 255);}.ace-tm .ace_comment.ace_doc.ace_tag {color: rgb(128, 159, 191);}.ace-tm .ace_constant.ace_numeric {color: rgb(0, 0, 205);}.ace-tm .ace_variable {color: rgb(49, 132, 149);}.ace-tm .ace_xml-pe {color: rgb(104, 104, 91);}.ace-tm .ace_entity.ace_name.ace_function {color: #0000A2;}.ace-tm .ace_heading {color: rgb(12, 7, 255);}.ace-tm .ace_list {color:rgb(185, 6, 144);}.ace-tm .ace_meta.ace_tag {color:rgb(0, 22, 142);}.ace-tm .ace_string.ace_regex {color: rgb(255, 0, 0)}.ace-tm .ace_marker-layer .ace_selection {background: rgb(181, 213, 255);}.ace-tm.ace_multiselect .ace_selection.ace_start {box-shadow: 0 0 3px 0px white;}.ace-tm .ace_marker-layer .ace_step {background: rgb(252, 255, 0);}.ace-tm .ace_marker-layer .ace_stack {background: rgb(164, 229, 101);}.ace-tm .ace_marker-layer .ace_bracket {margin: -1px 0 0 -1px;border: 1px solid rgb(192, 192, 192);}.ace-tm .ace_marker-layer .ace_active-line {background: rgba(0, 0, 0, 0.07);}.ace-tm .ace_gutter-active-line {background-color : #dcdcdc;}.ace-tm .ace_marker-layer .ace_selected-word {background: rgb(250, 250, 255);border: 1px solid rgb(200, 200, 250);}.ace-tm .ace_indent-guide {background: url("data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAACCAYAAACZgbYnAAAAE0lEQVQImWP4////f4bLly//BwAmVgd1/w11/gAAAABJRU5ErkJggg==") right repeat-y;}
/*# sourceURL=ace/css/ace-tm */
      </style>
      <style>
            .error_widget_wrapper {        background: inherit;        color: inherit;        border:none    }    .error_widget {        border-top: solid 2px;        border-bottom: solid 2px;        margin: 5px 0;        padding: 10px 40px;        white-space: pre-wrap;    }    .error_widget.ace_error, .error_widget_arrow.ace_error{        border-color: #ff5a5a    }    .error_widget.ace_warning, .error_widget_arrow.ace_warning{        border-color: #F1D817    }    .error_widget.ace_info, .error_widget_arrow.ace_info{        border-color: #5a5a5a    }    .error_widget.ace_ok, .error_widget_arrow.ace_ok{        border-color: #5aaa5a    }    .error_widget_arrow {        position: absolute;        border: solid 5px;        border-top-color: transparent!important;        border-right-color: transparent!important;        border-left-color: transparent!important;        top: -5px;    }
      </style>
    </head>
    <body
      class="euiBody-hasOverlayMask"
    >
      <div />
      <div
        class="euiOverlayMask euiOverlayMask--aboveHeader"
      >
        <div
          data-eui="EuiFocusTrap"
        >
          <div
            class="euiModal euiModal--maxWidth-default savedObjectsModal"
            data-test-subj="savedObjectsDuplicateModal"
            tabindex="0"
          >
            <button
              aria-label="Closes this modal window"
              class="euiButtonIcon euiButtonIcon--text euiButtonIcon--empty euiButtonIcon--xSmall euiModal__closeIcon"
              type="button"
            >
              <span
                aria-hidden="true"
                class="euiButtonIcon__icon"
                color="inherit"
                data-euiicon-type="cross"
              />
            </button>
            <div
              class="euiModal__flex"
            >
              <div
                class="euiModalHeader"
              >
                <div
                  class="euiModalHeader__title"
                >
                  <span>
                    Copy 3 objects to another workspace?
                  </span>
                </div>
              </div>
              <div
                class="euiModalBody"
              >
                <div
                  class="euiModalBody__overflow"
                >
                  <div
                    class="euiFormRow euiFormRow--fullWidth"
                    id="generated-id-row"
                  >
                    <div
                      class="euiFormRow__labelWrapper"
                    >
                      <label
                        class="euiFormLabel euiFormRow__label"
                        for="generated-id"
                      >
                        Workspace
                      </label>
                    </div>
                    <div
                      class="euiFormRow__fieldWrapper"
                    >
                      <div
                        class="euiText euiText--small"
                      >
                        <div
                          class="euiTextColor euiTextColor--subdued"
                        >
                          Move copied saved  objects to the selected workspace.
                        </div>
                      </div>
                      <div
                        class="euiSpacer euiSpacer--s"
                      />
                      <div
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        class="euiComboBox euiComboBox-isInvalid"
                        role="combobox"
                      >
                        <div
                          class="euiFormControlLayout"
                        >
                          <div
                            class="euiFormControlLayout__childrenWrapper"
                          >
                            <div
                              class="euiComboBox__inputWrap euiComboBox__inputWrap--noWrap"
                              data-test-subj="comboBoxInput"
                              tabindex="-1"
                            >
                              <p
                                class="euiComboBoxPlaceholder"
                              >
                                Select a workspace
                              </p>
                              <div
                                class="euiComboBox__input"
                                style="font-size: 14px; display: inline-block;"
                              >
                                <input
                                  aria-controls=""
                                  data-test-subj="comboBoxSearchInput"
                                  role="textbox"
                                  style="box-sizing: content-box; width: 2px;"
                                  value=""
                                />
                                <div
                                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                                />
                              </div>
                            </div>
                            <div
                              class="euiFormControlLayoutIcons euiFormControlLayoutIcons--right"
                            >
                              <button
                                aria-label="Open list of options"
                                class="euiFormControlLayoutCustomIcon euiFormControlLayoutCustomIcon--clickable"
                                data-test-subj="comboBoxToggleListButton"
                                type="button"
                              >
                                <span
                                  aria-hidden="true"
                                  class="euiFormControlLayoutCustomIcon__icon"
                                  data-euiicon-type="arrowDown"
                                />
                              </button>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="euiSpacer euiSpacer--l"
                  />
                  <div
                    class="euiFormRow euiFormRow--fullWidth"
                    id="generated-id-row"
                  >
                    <div
                      class="euiFormRow__labelWrapper"
                    >
                      <label
                        class="euiFormLabel euiFormRow__label"
                        for="generated-id"
                      >
                        Copy related objects
                      </label>
                    </div>
                    <div
                      class="euiFormRow__fieldWrapper"
                    >
                      <div
                        class="euiSpacer euiSpacer--s"
                      />
                      <div
                        class="euiCheckbox"
                      >
                        <input
                          checked=""
                          class="euiCheckbox__input"
                          id="includeReferencesDeep"
                          type="checkbox"
                        />
                        <div
                          class="euiCheckbox__square"
                        />
                        <label
                          class="euiCheckbox__label"
                          for="includeReferencesDeep"
                        >
                          Copy the selected object and any related objects (recommended).
                        </label>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="euiModalFooter"
              >
                <button
                  class="euiButtonEmpty euiButtonEmpty--primary"
                  data-test-subj="duplicateCancelButton"
                  type="button"
                >
                  <span
                    class="euiButtonContent euiButtonEmpty__content"
                  >
                    <span
                      class="euiButtonEmpty__text"
                    >
                      <span>
                        Cancel
                      </span>
                    </span>
                  </span>
                </button>
                <button
                  class="euiButton euiButton--primary euiButton--fill euiButton-isDisabled"
                  data-test-subj="duplicateConfirmButton"
                  disabled=""
                  type="button"
                >
                  <span
                    class="euiButtonContent euiButton__content"
                  >
                    <span
                      class="euiButton__text"
                    >
                      <span>
                        Copy
                      </span>
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </body>
  </html>,
]
`;
